cloudbridge/cloud/providers/gce/provider.py

```diff
@@ -157,6 +160,38 @@ def parse_url(self, url):
                 out.parameters[parameter] = m.group(index + 1)
             return out
 
+    def get_resource_url_with_default(self, resource, identifier):
```

Comment:
After this PR resource ID should mean resource URL everywhere and the last parameter of this method should be resource `name`, right?

Reply:
Yes. After this PR,  resource id will be resource URL (GCE selfLink), except where a resource is not mapped to a real GCE service (such as GCESecurityGroup).